example-client/client.py

```diff
@@ -67,40 +69,42 @@ def run_structure(input: str) -> Optional[str]:
     # Runs are asynchronous, so we need to poll the status until it's no longer running.
     structure_run_id = structure_run["structure_run_id"]
     status = structure_run["status"]
-    printed_event_ids = set()  # Keep track of which events we've printed.
-    while status not in ("SUCCEEDED", "FAILED"):
-        structure_run = get_structure_run(
-            host=HOST, api_key=GT_API_KEY, run_id=structure_run_id
+    offset = 0
+    output = None
+    while not is_status_complete(status) or output is None:
```

Comment:
I think we can do what we had in the Draft PR, right? Or what's the advantage of querying status in addition to the event-driven logic?

Reply:
the status is coming from the system events. its a helper function to determine if the run is in a completed status.

Reply:
I don't think we need to be checking BOTH status AND output. We can query a bool to see if we've hit a terminal state, no? It seems like you're emulating the status approach here, when we want to eliminate doing both status and events.

Reply:
> We can query a bool to see if we've hit a terminal state, no?

i'm not exactly sure what this means. when the structure has a new status, it comes through as an event. for this specific example, we can remove the check of status events and just check for the `FinishStructureRunEvent` to get the output, but there are other scenarios where you would want to check for the status events as well.